File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cs
public class Solution {
    public int MaxDepth(string s) {
        short ans = 0;
        Stack <char> st = new Stack <char>();
        for (int i = 0;i < s.Length; i++){
            if (st.Count == 0){
                if (s[i] == '(')
                    st.Push(s[i]);
            }
            else if (s[i] == '(')
                st.Push(s[i]);
            
            else if (s[i] == ')' && st.Peek() == '('){
                short sz = (short)st.Count;
                ans = Math.Max(ans , sz);    
                st.Pop();
            }
        }
        
        return ans;
    }
}

File: 1614-Maximum-Nesting-Depth-of-the-Parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack <char> st;
        short ans = 0;
        for (int i = 0; i < s.size(); i++){
            if (st.empty()){
                if (s[i] == '(')
                    st.push(s[i]);
            }
            else if (s[i] == '(') st.push(s[i]);
            else if (st.top() == '(' && s[i] == ')'){
                short sz = st.size();
                ans = max(sz , ans);
                st.pop();
            }    
        }
        return ans;
    }
};